### PR TITLE
Add the tab-completion support of SERVICE_LEVEL statements

### PR DESCRIPTION
**Pull request by fruch**
Fixes: #43

### TODOs
- [x]  - add test for enterprise in github action
- [x] - add unit test for SLA

### CODE REVIEW

**Comment by fruch** [comment]
@eliransin want to take it for a test ride ?

File: pylib/cqlshlib/cql3handling.py
Comment (juliayakovlev) [inline pylib/cqlshlib/cql3handling.py:1514]: create statement has "if not exists" option:
https://enterprise.docs.scylladb.com/stable/using-scylla/workload-prioritization.html#:~:text=%27primary%27).-,Syntax,-CREATE%20SERVICE_LEVEL%20%5BIF

File: pylib/cqlshlib/cql3handling.py
Comment (juliayakovlev) [inline pylib/cqlshlib/cql3handling.py:1532]: another options: 
`ATTACH SERVICE_LEVEL <service_level_name> TO <role_name>;`

`LIST ALL ATTACHED SERVICE_LEVELS;`

`LIST ATTACHED SERVICE_LEVEL OF <role_name>;`

`DETACH SERVICE_LEVEL FROM <role_name>;`

File: pylib/cqlshlib/cql3handling.py
Comment (juliayakovlev) [inline pylib/cqlshlib/cql3handling.py:302]: there are `detach` and `attach` statements

File: pylib/cqlshlib/cql3handling.py
Comment (juliayakovlev) [inline pylib/cqlshlib/cql3handling.py:1524]: SHARES is `integer`

**Review by juliayakovlev — CHANGES_REQUESTED** [review]
there are few options/statements that should be added

File: pylib/cqlshlib/cql3handling.py
Comment (fruch) [reply pylib/cqlshlib/cql3handling.py:1540]: added those

File: pylib/cqlshlib/cql3handling.py
Comment (fruch) [reply pylib/cqlshlib/cql3handling.py:302]: added those

File: pylib/cqlshlib/cql3handling.py
Comment (fruch) [reply pylib/cqlshlib/cql3handling.py]: added the optional if not exists

File: pylib/cqlshlib/cql3handling.py
Comment (fruch) [reply pylib/cqlshlib/cql3handling.py]: fix to a whole number, i.e. share can't be negative, right ?

File: pylib/cqlshlib/cql3handling.py
Comment (Lorak-mmk) [inline pylib/cqlshlib/cql3handling.py:1525]: where are those documented? On https://enterprise.docs.scylladb.com/stable/using-scylla/workload-prioritization.html I only see `SHARES`

File: pylib/cqlshlib/cql3handling.py
Comment (Lorak-mmk) [inline pylib/cqlshlib/cql3handling.py:1532]: In the docs I also see different variants: https://enterprise.docs.scylladb.com/stable/using-scylla/workload
-  `LIST SERVICE_LEVEL <name>;`
- `LIST ATTACHED SERVICE_LEVEL OF <role>;`

This grammar also doesn't support `LIST ALL SERVICE_LEVELS`, it requires `ATTACHED`, doesn't it? Version without `ATTACHED` is also mentioned in the docs.

**Review by Lorak-mmk — CHANGES_REQUESTED** [review]

File: pylib/cqlshlib/cql3handling.py
Comment (juliayakovlev) [reply pylib/cqlshlib/cql3handling.py]: From documnetation:
```
shares_number - The number of shares of the resources you’re granting to the service level name. You can use any number within the range from 1 to 1000
```

File: pylib/cqlshlib/cql3handling.py
Comment (juliayakovlev) [reply pylib/cqlshlib/cql3handling.py:1525]: @Lorak-mmk You can find it [here](https://opensource.docs.scylladb.com/stable/using-scylla/workload-attributes.html)

File: pylib/cqlshlib/cql3handling.py
Comment (juliayakovlev) [reply pylib/cqlshlib/cql3handling.py:1532]: @Lorak-mmk 

`LIST ALL SERVICE_LEVELS` is supported:
https://enterprise.docs.scylladb.com/stable/using-scylla/workload-prioritization.html#:~:text=service%20level%20allocations%3A-,LIST%20ALL%20SERVICE_LEVELS%3B,-service_level%20%7C%20shares

**Review by juliayakovlev — APPROVED** [review]

File: pylib/cqlshlib/cql3handling.py
Comment (fruch) [reply pylib/cqlshlib/cql3handling.py:1525]: it's a part in OSS part in enterprise

File: pylib/cqlshlib/cql3handling.py
Comment (Lorak-mmk) [reply pylib/cqlshlib/cql3handling.py:1532]: > @Lorak-mmk
> 
> `LIST ALL SERVICE_LEVELS` is supported: https://enterprise.docs.scylladb.com/stable/using-scylla/workload-prioritization.html#:~:text=service%20level%20allocations%3A-,LIST%20ALL%20SERVICE_LEVELS%3B,-service_level%20%7C%20shares

I meant that it is supported by Scylla, but grammar in this PR does not seem to support it, unless I'm reading it wrong.
Grammar in this PR also seems to not support `LIST SERVICE_LEVEL <name>;` and `LIST ATTACHED SERVICE_LEVEL OF <role>;`, both variants are supported by Scylla Enterprise.

Comment (Lorak-mmk) [reply pylib/cqlshlib/cql3handling.py:1532]: @fruch ?

Comment (Lorak-mmk) [reply pylib/cqlshlib/cql3handling.py:1532]: ping

**Comment by roydahan** [comment]
@fruch I guess you're still waiting a review here...

@Lorak-mmk could you please review?

Who is the main maintainer of this repository?

**Comment by Lorak-mmk** [comment]
> @fruch I guess you're still waiting a review here...
> 
> @Lorak-mmk could you please review?
> 
> Who is the main maintainer of this repository?

I did review and I'm waiting for a response to my comment, I pinged two times.

**Comment by roydahan** [comment]
Thanks, weird that your review appears as dismissed, but no tracking of why.

**Comment by roydahan** [comment]
@fruch ?

File: pylib/cqlshlib/cql3handling.py
Comment (fruch) [reply pylib/cqlshlib/cql3handling.py]: now it's addressed, and there a unit test to cover it

**Pull request merged**
